scivision/io/reader.py

```diff
@@ -29,22 +29,28 @@ def _parse_url(path: os.PathLike, branch: str = "main"):
     return parsed.geturl()
 
 
-def _parse_config(path: os.PathLike, branch: str = "main") -> dict:
-    """Parse the `scivision` config file."""
-    # check that this is a path to a yaml file
-    if not path.endswith((".yml", ".yaml",)):
-        raise ValueError(f"Invalid configuration filename: {path}")
+def _parse_config(path: os.PathLike, branch: str = "main", model_config=True) -> dict:
+    """Parse the scivision.yml file from a GitHub repository.
+    Will also accept differently named yaml if a full path provided or a local file.
+    """
 
     if _is_url(path):
-        path = _parse_url(path)
+        path = _parse_url(path, branch)
+
+    # check that this is a path to a yaml file
+    # if not, assume it is a repo containing "scivision.yml"
+    if not path.endswith((".yml", ".yaml",)):
+        path = path + "scivision.yml"
 
+    # This will throw an error if the path does not exist
     file = fsspec.open(path)
     with file as config_file:
         stream = config_file.read()
         config = yaml.safe_load(stream)
 
-    # make sure we at least have an input to the function
-    assert "X" in config["prediction_fn"]["args"].keys()
+    # make sure a model at least has an input to the function
+    if model_config:
+        assert "X" in config["prediction_fn"]["args"].keys()
```

Comment:
See comment below, but might be cleaner to handle data and model in separate functions, if they are different (rather than a flag)

Reply:
@ots22 you're right in the comment below, refactor satisfies both your comments

Reply:
actually I should probably add a test for `load_pretrained_model` before merging this

Reply:
LGTM!